src/library/aoide/domain/cue.cpp

```diff
@@ -0,0 +1,148 @@
+#include "library/aoide/domain/cue.h"
+
+#include "util/assert.h"
+#include "util/logger.h"
+
+namespace {
+
+const mixxx::Logger kLogger("aoide Cue");
+
+const int kPositionMarkerCustom = 0;
+const int kPositionMarkerLoad = 1;
+const int kPositionMarkerMain = 2;
+const int kPositionMarkerIntro = 3;
+const int kPositionMarkerOutro = 4;
+const int kPositionMarkerJump = 5;
+const int kPositionMarkerLoop = 6;
+const int kPositionMarkerSample = 7;
+
+} // anonymous namespace
+
+AoidePositionMarker::AoidePositionMarker(const Cue& cue, double cuePositionToMillis) {
+    switch (cue.getType()) {
+    case Cue::CUE:
+    case Cue::LOAD:
+        setStart(cue.getPosition() * cuePositionToMillis);
```

Comment:
Not possible, because unfortunately cue position and length are measured in sample offsets, assuming that channel samples are interleaved. The latter is one of the main design flaws that complicates the engine code in so many places. Moreover, these cue properties depend on the context, namely the sample rate and channel count of the PCM signal.

```
cuePositionToMillis = 1000.0 / (sampleRate * channelCount)
```

aoide fixes this by storing the values in plain time units, independent from the properties of the underlying PCM signal.

Reply:
Ha, my bad. Can you explain this inline.

Reply:
I have added a comment in the `TrackExporter` where this value is calculated. Cue points of tracks with an invalid sample rate or number of channels cannot be exported. It only happens for corrupt or unsupported files.